### PR TITLE
add formatter for gasPrice and balance

### PR DESCRIPTION
**Pull request by frozeman**

### CODE REVIEW

File: dist/ethereum.js [hunk 1]
```diff
@@ -310,13 +310,13 @@ var exportNatspecGlobals = function (vars) {
...
     contract.call = function (options) {
-        contract._isTransact = false;
+        contract._isTransaction = false;
         contract._options = options;
         return contract;
     };
 
-    contract.transact = function (options) {
-        contract._isTransact = true;
+    contract.sendTransaction = function (options) {
```
Comment (debris) [inline dist/ethereum.js]: can `contract.transact` be also marked as deprecated?

File: dist/ethereum.js [hunk 2]
```diff
@@ -452,24 +452,40 @@ var addEventsToContract = function (contract, desc, address) {
...
+ * @param abi - abi json description of the contract, which is being created
  * @returns contract object
  */
+var contract = function (abi) {
+
+    // return prototype
+    if(Object.prototype.toString.call(abi) === '[object Array]' && arguments.length === 1) {
+        return Contract.bind(this, abi);
+
+    // depreacted: auto initiate contract
```
Comment (debris) [inline dist/ethereum.js]: typo

File: lib/contract.js [hunk 3]
```diff
@@ -180,24 +180,40 @@ var addEventsToContract = function (contract, desc, address) {
...
+ * @param abi - abi json description of the contract, which is being created
  * @returns contract object
  */
+var contract = function (abi) {
+
+    // return prototype
+    if(Object.prototype.toString.call(abi) === '[object Array]' && arguments.length === 1) {
+        return Contract.bind(this, abi);
+
+    // depreacted: auto initiate contract
```
Comment (debris) [inline lib/contract.js]: typo

File: dist/ethereum.js (hunk 1 above)
Comment (frozeman) [inline dist/ethereum.js]: alright

File: lib/contract.js [hunk 4]
```diff
@@ -180,24 +188,40 @@ var addEventsToContract = function (contract, desc, address) {
...
- * @param address - address of the contract, which should be called
- * @param desc - abi json description of the contract, which is being created
+ * @param abi - abi json description of the contract, which is being created
  * @returns contract object
  */
+var contract = function (abi) {
 
-var contract = function (address, desc) {
+    // return prototype
+    if(Object.prototype.toString.call(abi) === '[object Array]' && arguments.length === 1) {
```
Comment (debris) [inline lib/contract.js]: can you make utils methods for checking types? eg.

```javascript
var isFunction = function (..) {}
var isArray = functin (...) {}
// and so on
```

also i would prefer to use `x instanceof Array`. Or you should check how underscore or lodash is checking if something is array.

File: lib/formatters.js [hunk 5]
```diff
@@ -99,6 +101,8 @@ var formatOutputInt = function (value) {
     return new BigNumber(value, 16);
 };
 
+
+
 /// Formats big right-aligned input bytes to uint
```
Comment (debris) [inline lib/formatters.js]: why are you adding blank lines here?

File: lib/formatters.js [hunk 6]
```diff
@@ -90,7 +90,9 @@ var signedIsNegative = function (value) {
 /// Formats input right-aligned input bytes to int
 /// @returns right-aligned input bytes formatted to int
 var formatOutputInt = function (value) {
+
     value = value || "0";
+
```
Comment (debris) [inline lib/formatters.js]: why are you adding blank lines here?

File: test/eth.methods.js [hunk 7]
```diff
@@ -8,7 +8,7 @@ describe('web3', function() {
         u.methodExists(web3.eth, 'getState');
         u.methodExists(web3.eth, 'getStorage');
         u.methodExists(web3.eth, 'getTransactionCount');
-        u.methodExists(web3.eth, 'getCode');
+        u.methodExists(web3.eth, 'getData');
```
Comment (debris) [inline test/eth.methods.js]: `getData` is not verbose enough for me. `getCode` implicitly told me, that I want to get it from some address. `getData` doesn't do that.

File: lib/contract.js [hunk 8]
```diff
@@ -180,24 +188,40 @@ var addEventsToContract = function (contract, desc, address) {
...
+        return Contract.bind(this, abi);
```
Comment (debris) [inline lib/contract.js]: it's not necessary to bind this. it can be also null

**Comment by debris** [comment]
generally it looks ok. please fix these few things that I mentioned. I also made jshint on master branch more strict. Please merge or rebase and try to apply style changes to your code (if it is necessary).

File: lib/contract.js (hunk 4 above)
Comment (frozeman) [inline lib/contract.js]: i changed to `instanceof Array`. I don't see the point for creating our own check functions here, if we really need that than lets use underscore etc. for its ok like that i say

File: lib/formatters.js (hunk 6 above)
Comment (frozeman) [inline lib/formatters.js]: visibility :)

**Pull request merged**